tensorflow/python/keras/engine/base_layer.py

```diff
@@ -1223,6 +1227,19 @@ def add_metric(self, value, aggregation=None, name=None):
       # Insert layers into the Keras Graph Network.
       self._graph_network_add_metric(value, aggregation, name)
 
+  @property
+  def callbacks(self):
+    return self._callbacks + self._gather_children_attribute('callbacks')
```

Comment:
`_gather_children_attribute` has been recently removed. Can you edit the PR to make it applicable? Thanks.

Reply:
Or now that I'm looking at it more, `self._flatten_layers()` looks like another option - maybe that's what you were talking about when you said the other method was removed?

Here it's used for `metrics`:
https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/keras/engine/base_layer.py#L1616-L1639